### DISCUSSION TITLE
How to separate unit and integration test

### DISCUSSION THREAD

**Discussion by Haberkamp** [vitest-dev/vitest#4675] [Q&A] [answered]
Hello, I wanted to ask if there's a way to separate unit and integration tests other than by specifying a glob pattern.

**Comment by AriPerkkio** [comment, marked as answer]
What kind of separation do you need? I would probably just specify two workspace projects and configure integration test related settings there. But this still requires glob patterns to separate tests from each other:

```ts
// vitest.config.ts
import { defineConfig } from "vitest/config";

export default defineConfig({
  test: {
    projects: [
      {
        test: {
          name: "unit",
          include: ["**/*.unit.test.ts"],
        },
      },
      {
        test: {
          name: "integration",
          include: ["**/*.integration.test.ts"],
          // More integration test related setup here...
        },
      },
    ],
  },
});
```

<details>
  <summary>Example with deprecated vitest.workspace.ts</summary>

```ts
// vitest.workspace.ts
import { defineWorkspace } from 'vitest/config';

export default defineWorkspace([
    {
        test: {
            name: 'unit',
            include: ['**/*.unit.test.ts'],
        },
    },
    {
        test: {
            name: 'integration',
            include: ['**/*.integration.test.ts'],
            // More integration test related setup here...
        },
    },
]);
```

</details>

**Reply by Frikki** [reply]
I don't see how this separates the runs. Doesn't Vitest run both? Or can I tell Vitest to run one and not the other?

**Reply by AriPerkkio** [reply]
You can use `--project unit` and `--project integration` flags to filter projects. https://vitest.dev/guide/cli.html#project

**Comment by vmasek** [comment]
I was also trying to figure it out and pretty good solution turned out to be separating integration tests into dedicated `tests` folder and grouping them with all the fixtures and related data/mocks.
Then having two config files one in the upper directory aside `src` and adding another `vitest.config.ts` file into the `tests` folder for the integration tests. In the nested one, you can specify all the setup files and configs and add one important property that will override `root` to be "one folder up" where other config file is (aside `src`).

This way the execution via Nx executor runs correctly for both targets and if I run individual tests via IntelliJ Idea or WebStorm, they also get picked up correctly as it prioritizes config closer to the file.
